apps/android/app/src/main/java/chat/simplex/app/views/chat/item/ChatItemView.kt

```diff
@@ -75,186 +79,246 @@ fun ChatItemView(
+          is CIContent.SndModerated -> MarkedDeletedItemView(cItem, cInfo.timedMessagesTTL, showMember = showMember)
+          is CIContent.RcvModerated -> MarkedDeletedItemView(cItem, cInfo.timedMessagesTTL, showMember = showMember)
+          is CIContent.InvalidJSON -> CIInvalidJSONView(c.json)
+        }
       }
 
-      when (val c = cItem.content) {
-        is CIContent.SndMsgContent -> ContentItem()
-        is CIContent.RcvMsgContent -> ContentItem()
-        is CIContent.SndDeleted -> DeletedItem()
-        is CIContent.RcvDeleted -> DeletedItem()
-        is CIContent.SndCall -> CallItem(c.status, c.duration)
-        is CIContent.RcvCall -> CallItem(c.status, c.duration)
-        is CIContent.RcvIntegrityError -> IntegrityErrorItemView(c.msgError, cItem, cInfo.timedMessagesTTL, showMember = showMember)
-        is CIContent.RcvDecryptionError -> CIRcvDecryptionError(c.msgDecryptError, c.msgCount, cItem, cInfo.timedMessagesTTL, showMember = showMember)
-        is CIContent.RcvGroupInvitation -> CIGroupInvitationView(cItem, c.groupInvitation, c.memberRole, joinGroup = joinGroup, chatIncognito = cInfo.incognito)
-        is CIContent.SndGroupInvitation -> CIGroupInvitationView(cItem, c.groupInvitation, c.memberRole, joinGroup = joinGroup, chatIncognito = cInfo.incognito)
-        is CIContent.RcvGroupEventContent -> CIEventView(cItem)
-        is CIContent.SndGroupEventContent -> CIEventView(cItem)
-        is CIContent.RcvConnEventContent -> CIEventView(cItem)
-        is CIContent.SndConnEventContent -> CIEventView(cItem)
-        is CIContent.RcvChatFeature -> CIChatFeatureView(cItem, c.feature, c.enabled.iconColor)
-        is CIContent.SndChatFeature -> CIChatFeatureView(cItem, c.feature, c.enabled.iconColor)
-        is CIContent.RcvChatPreference -> {
-          val ct = if (cInfo is ChatInfo.Direct) cInfo.contact else null
-          CIFeaturePreferenceView(cItem, ct, c.feature, c.allowed, acceptFeature)
-        }
-        is CIContent.SndChatPreference -> CIChatFeatureView(cItem, c.feature, MaterialTheme.colors.secondary, icon = c.feature.icon,)
-        is CIContent.RcvGroupFeature -> CIChatFeatureView(cItem, c.groupFeature, c.preference.enable.iconColor)
-        is CIContent.SndGroupFeature -> CIChatFeatureView(cItem, c.groupFeature, c.preference.enable.iconColor)
-        is CIContent.RcvChatFeatureRejected -> CIChatFeatureView(cItem, c.feature, Color.Red)
-        is CIContent.RcvGroupFeatureRejected -> CIChatFeatureView(cItem, c.groupFeature, Color.Red)
-        is CIContent.SndModerated -> MarkedDeletedItemView(cItem, cInfo.timedMessagesTTL, showMember = showMember)
-        is CIContent.RcvModerated -> MarkedDeletedItemView(cItem, cInfo.timedMessagesTTL, showMember = showMember)
-        is CIContent.InvalidJSON -> CIInvalidJSONView(c.json)
+      if (cItem.content.msgContent != null && cItem.meta.itemDeleted == null && cItem.reactions.isNotEmpty()) {
+        ChatItemReactions()
```

Comment:
padding between voice messages and reactions is very large, reactions are even closer to the next message, same on iOS